src/realm/sync/subscriptions.cpp

```diff
@@ -352,6 +352,18 @@ void MutableSubscriptionSet::insert_sub(const Subscription& sub)
     m_subs.push_back(sub);
 }
 
+void MutableSubscriptionSet::on_boostrap_cleared()
+{
+    update_state(State::Pending);
```

Comment:
> How is setting the subscription state back to pending helping?

Fair point, setting back to the pending state isn't actually necessary. I can revert this. The main thing is that we want this subscription set to be picked up to send to the server again. This is done by `get_next_pending_version()` which queries for "(pending OR boostrapping) and snapshot_version > client_upload_version" so the only relevant change is to update the snapshot version at this point so that we restart the bootstrap process of this subscription with the server.